homeassistant/components/watttime/sensor.py

```diff
@@ -0,0 +1,142 @@
+    CONF_BALANCING_AUTHORITY,
+    CONF_BALANCING_AUTHORITY_ABBREV,
+    CONF_BALANCING_AUTHORITY_ID,
+    DATA_COORDINATOR,
+    DOMAIN,
+)
+
+ATTR_BALANCING_AUTHORITY = "balancing_authority"
+ATTR_BALANCING_AUTHORITY_ABBREV = "balancing_authority_abbreviation"
+ATTR_BALANCING_AUTHORITY_ID = "balancing_authority_id"
+
+DEFAULT_ATTRIBUTION = "Pickup data provided by WattTime"
+
+SENSOR_TYPE_REALTIME_EMISSIONS_MOER = "realtime_emissions_moer"
+SENSOR_TYPE_REALTIME_EMISSIONS_PERCENT = "realtime_emissions_percent"
+
+
+@dataclass
+class RealtimeEmissionsSensorDescriptionMixin:
+    """Define an entity description mixin for realtime emissions sensors."""
+
+    data_key: str
+
+
+@dataclass
+class RealtimeEmissionsSensorEntityDescription(
+    SensorEntityDescription, RealtimeEmissionsSensorDescriptionMixin
+):
+    """Describe a realtime emissions sensor."""
+
+
+REALTIME_EMISSIONS_SENSOR_DESCRIPTIONS = (
+    RealtimeEmissionsSensorEntityDescription(
+        key=SENSOR_TYPE_REALTIME_EMISSIONS_MOER,
+        name="Marginal Operating Emissions Rate",
+        icon="mdi:blur",
+        native_unit_of_measurement="lbs CO2/MWh",
+        state_class=STATE_CLASS_MEASUREMENT,
+        data_key="moer",
+    ),
+    RealtimeEmissionsSensorEntityDescription(
+        key=SENSOR_TYPE_REALTIME_EMISSIONS_PERCENT,
+        name="Relative Marginal Emissions Intensity",
+        icon="mdi:blur",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        data_key="percent",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up WattTime sensors based on a config entry."""
+    coordinator = hass.data[DOMAIN][DATA_COORDINATOR][entry.entry_id]
+    async_add_entities(
+        [
+            RealtimeEmissionsSensor(coordinator, description)
+            for description in REALTIME_EMISSIONS_SENSOR_DESCRIPTIONS
+            if description.data_key in coordinator.data
+        ]
+    )
+
+
+class RealtimeEmissionsSensor(CoordinatorEntity, SensorEntity):
+    """Define a realtime emissions sensor."""
+
+    entity_description: RealtimeEmissionsSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        description: RealtimeEmissionsSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+
+        if TYPE_CHECKING:
+            assert coordinator.config_entry
+
+        ba_abbrev = coordinator.config_entry.data[CONF_BALANCING_AUTHORITY_ABBREV]
+
+        self._attr_extra_state_attributes = {
+            ATTR_ATTRIBUTION: DEFAULT_ATTRIBUTION,
+            ATTR_BALANCING_AUTHORITY: coordinator.config_entry.data[
+                CONF_BALANCING_AUTHORITY
+            ],
+            ATTR_BALANCING_AUTHORITY_ABBREV: ba_abbrev,
+            ATTR_BALANCING_AUTHORITY_ID: coordinator.config_entry.data[
```

Comment:
Why are these data points important ?

Reply:
The abbreviation is part of the entity name. The name/ID could be useful for navigating their website, but if you'd rather not pollute the state machine, I can remove. 👍🏻